src/cephadm/tests/test_container_engine.py

```diff
@@ -0,0 +1,57 @@
+from unittest import mock
+
+import pytest
+
+from tests.fixtures import with_cephadm_ctx
+
+with mock.patch("builtins.open", create=True):
+    from importlib.machinery import SourceFileLoader
+
+    cephadm = SourceFileLoader("cephadm", "cephadm.py").load_module()
```

Comment:
can you give me a brief explanation of what this part is for?

Reply:
This is (mostly) a copy and paste of the code that exists in the other test files. To the best of my knowledge it's doing two things: first, it mocks out open [1] and then it loads cephadm as a python module.  I am guessing that (some) of this is unnecessary now that `cephadm.py` has replaced `cephadm`. I'm happy to look into it in depth. However, as far as this PR goes I'd just ask that we  don't worry about it too much :-)

[1] - I assume, but have not verified, that there are module level calls to open. This can be "fixed" in the future, but for now it may be necessary to import cephadm into the test module(s).

Reply:
It appears that the (mock) patching of open was added in 81a7df0498dc649c68971ca31b9ae60874f61133. It appears to me to be related to how logging was set up in a ` if __name__ == "__main__":` block.
However, this may not be needed any more because this logic has been moved. Currently the entire contents of the ` if __name__ == "__main__":` block is a call to `main()`.


In my new file I can change the entire block to `import cephadm`. It appears to work.
What I'm now considering is to add a utility function, `import_cephadm` that can be added to fixtures.py. This function can then be used to import cephadm and do any importlib trickery if needed.  Thoughts?

Reply:
OK, it's a smidge more complex than that due to how fixtures is also importing cephadm, and the open mock appears to still be needed (for test_cephadm.py at least). Let my first request stand: allow me this copy-n-paste for this PR.  Later, I will follow up with another PR that investigates and tries to clean up the cephadm import business in these test files.

Reply:
works for me, thanks for the analysis.